### ISSUE TITLE
Feature Request: Merge two zerolog Events together

### ISSUE THREAD

**Issue by aaomidi** [rs/zerolog#292]
Say I want to create a function for a struct that returns an Event - this doesn't get immediately logged but contains boilerplate of all the fields I want to log when it comes to that struct. 

Example:

```go
// Aggregation represents an aggregation task
type Aggregation struct {
	TraceID string `json:"trace-id"`
	AggregationID string `json:"aggregation-id"`
}
func (a Aggregation) LogEvent() *zerolog.Event {
	event := zerolog.Event{}
	return event.
		Str("trace ID", a.TraceID).
		Str("aggregation ID", a.AggregationID)
	/// etc
}
```

I then want to consume this "event" where I will be actually logging. Something like:

```
log.Info().WithEvent(a.LogEvent).Str("Some Other Key", "Some Other Value").Msg("Some log message")
```

**Comment by Hovercross** [comment]
This would be a great feature for me if it isn't already possible. I've done something similar in Zap with some error handling, where we created our own error wrapper struct, and if you were passing the error up the chain you could annotate it with additional fields. When it was time to either handle or log the error, you could log it with all the fields that had been passed up the stack with the error - including a stack trace from when our code first saw the error. I find Zerolog's interface a bit easier to use and prefer it in my personal projects, and I'd love to be able to create a similar interface to wrap errors like that.

**Comment by irl-segfault** [comment]
Makes sense.  Maybe using `Dict()` would work?  you could do something like `log.Dict("error_annotations", err.Dict())` or something similar if you implement a custom error type to do that work.

otherwise, ZapWrap could stuff a sublogger in the error itself, and then handleOrLog() grabs the logger out of the error (which by the union of all the fields) and do the logging.
